cloud-startup.sh

```diff
@@ -20,28 +31,86 @@ done
+    echo "Failed to create/access database ${MYSQL_DATABASE}"
+    exit 1
+fi
+
+echo "Database ${MYSQL_DATABASE} ready"
+
+# Run database migrations using alembic
+# This ensures all database tables and schemas are up to date
+cd /app
 alembic upgrade head
 
-# check backend input parameters
+# Initialize admin user and organization
+# Only proceeds if all required environment variables are set
+# This section handles first-time setup of the application
+if [ ! -z "$INITIAL_FIREBASE_UID" ] && [ ! -z "$INITIAL_USER_NAME" ] && [ ! -z "$INITIAL_USER_EMAIL" ]; then
+    echo "Checking for existing admin user..."
+    # Check if user already exists to prevent duplicate creation
+    USER_EXISTS=$(mysql -h "$MYSQL_SERVER" -u "$MYSQL_USER" -p"$MYSQL_PASSWORD" -N -s -e \
+        "SELECT COUNT(*) FROM ${MYSQL_DATABASE}.users WHERE uid='$INITIAL_FIREBASE_UID';")
+
+    if [ "$USER_EXISTS" -eq "0" ]; then
+        # Create default organization first (required due to foreign key constraint)
+        echo "Creating default organization..."
+        mysql -h "$MYSQL_SERVER" -u "$MYSQL_USER" -p"$MYSQL_PASSWORD" ${MYSQL_DATABASE} <<-EOSQL
+            INSERT IGNORE INTO organization (id, name) VALUES (1, 'Default Organization');
+EOSQL
+
+        # Create the initial admin user
+        echo "Creating initial admin user..."
+        mysql -h "$MYSQL_SERVER" -u "$MYSQL_USER" -p"$MYSQL_PASSWORD" ${MYSQL_DATABASE} <<-EOSQL
+            INSERT INTO users (uid, organization_id, name, email, active)
+            VALUES ('$INITIAL_FIREBASE_UID', 1, '$INITIAL_USER_NAME', '$INITIAL_USER_EMAIL', true);
+EOSQL
+        echo "Initial admin user created successfully"
+    else
+        echo "Admin user already exists, skipping creation"
+    fi
+fi
+
+# Verify backend configuration
+# Ensures required environment variables are set before starting the application
 echo "Host: $BACKEND_HOST"
 echo "Port: $BACKEND_PORT"
-
 if [ -z "$BACKEND_HOST" ] || [ -z "$BACKEND_PORT" ]; then
     echo "Please provide 'BACKEND_HOST' and 'BACKEND_PORT' environment variables"
     exit 1
 fi
 
-# Start the application in the background
+# Start the application in background
+echo "Starting application..."
 poetry run python main.py --host="$BACKEND_HOST" --port="$BACKEND_PORT" &
 APP_PID=$!
 
-# Check load balancer
+# Allow initial startup time matching ECS health check startPeriod
+echo "Waiting for initial startup..."
+sleep 30
+
+# Single initial health check before load balancer check
+echo "Verifying initial health..."
+if ! curl -v http://127.0.0.1:8000/health; then
+    echo "Initial health check failed"
+    # Don't exit - let ECS handle it
+fi
+
+# Load balancer health check function
+# Verifies that the application is accessible through the load balancer
 check_load_balancer() {
     if [ -n "$AWS_SERVICE_URL" ]; then
         echo "Checking load balancer status..."
         max_tries=30
         counter=0
+        # Try for 5 minutes (30 attempts * 10 seconds)
```

Comment:
A minor point, but while we're at it, wouldn't it be nice to have `10 seconds` as a variable as well?